src/js/load-package.ts

```diff
@@ -162,17 +162,21 @@ async function downloadPackage(
   name: string,
   channel: string
 ): Promise<Uint8Array> {
-  let file_name;
+  let file_name, file_sub_resource_hash;
   if (channel === DEFAULT_CHANNEL) {
     if (!(name in API.packages)) {
       throw new Error(`Internal error: no entry for package named ${name}`);
     }
     file_name = API.packages[name].file_name;
+    file_sub_resource_hash = API.package_loader.sub_resource_hash(
+      API.packages[name].sha_256
```

Comment:
Unrelated to this PR, but it should be `sha256` without the `_` in the JSON to be consistent with https://docs.conda.io/projects/conda-build/en/latest/concepts/generating-index.html#repodata-json Sorry I missed it in your first PR.